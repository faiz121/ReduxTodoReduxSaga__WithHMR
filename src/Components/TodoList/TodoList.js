import React from 'react'
import TodoItem from '../TodoItem'
import { connect } from 'react-redux'
import { deleteTodoDB } from '../../Action'
import todosSelector from '../../Selector/filterTodoSelector'

const Todo = (props) => {
    const { todos } = props;
    return (
        <table className="table table-xs">
          <thead className="thead-default">
          <tr>
            <th>To-do</th>
            <th>Delete</th>
          </tr>
          </thead>
          <tbody>
          {todos.map(todo => {
                return (
                    <TodoItem key={todo._id} todo={todo.task} todoId={todo._id} remove={()=> {
                      props.deleteTodoDB(todo._id);
                     }
                    }
                />
                )
              })}
          </tbody>
        </table>
    )
};

const  mapStateToProps = (state) => {
  return {
    todos: todosSelector(state),
  };
};
export const Unwrapped = Todo;
export default connect(mapStateToProps, {deleteTodoDB})(Todo);
