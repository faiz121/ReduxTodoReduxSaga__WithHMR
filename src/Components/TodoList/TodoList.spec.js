import React from 'react';
import {Provider} from 'react-redux';
import store from '../../Store'
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import {Unwrapped as UnwrappedTodo } from './TodoList'
const todos = [
  {
    _id: '1',
    task: 'Do that',

  }
];
test('TodoList take a snapshot', () => {
  const component = shallow(<Provider store={store}><UnwrappedTodo
      deleteTodoDB={()=>{}}
      todos={todos} /></Provider>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
