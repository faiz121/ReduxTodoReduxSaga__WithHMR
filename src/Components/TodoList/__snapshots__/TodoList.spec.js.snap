// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList take a snapshot 1`] = `
<Todo
  deleteTodoDB={[Function]}
  todos={
    Array [
      Object {
        "_id": "1",
        "task": "Do that",
      },
    ]
  }
/>
`;
